### DISCUSSION TITLE
Visual representation of mutations.

### DISCUSSION THREAD

**Discussion by cmnrd** [lf-lang/lingua-franca#2042] [Ideas]
Currently, reactions and mutations are rendered using the same shape and style. The following program
```
target Cpp

main reactor {
    reaction react(startup)
    mutation mutate(shutdown)
}
```
produces this diagram:
![Mutation](https://github.com/lf-lang/lingua-franca/assets/6460123/f6b91512-c00b-4b70-b326-5c4c9449964c)

I think the two concepts should also be distinguished visually. I would like to get some more opinions and proposal on how mutations could be visualized.

Here are some proposals based on yesterday's discussion:

### Making mutations darker: 
  ![Mutation](https://github.com/lf-lang/lingua-franca/assets/6460123/de2d34d8-c774-4600-81af-d739cd66daa6)
### Using an hourglass shape:
  ![Mutation](https://github.com/lf-lang/lingua-franca/assets/6460123/f848009b-7397-4a42-8918-399a51c70e5f)
### Using a one-sided arrow:
  ![Mutation](https://github.com/lf-lang/lingua-franca/assets/6460123/6c710ada-af27-4edf-b60f-f069911409dc)
### Using a rhomboid:
  ![Mutation](https://github.com/lf-lang/lingua-franca/assets/6460123/3ff82947-1dd1-4074-851f-369f72553681)

I added the rhomboid as an additional option, which I think looks nice and is clearly distinguishable from reactions. What do you think?

**Comment by soerendomroes** [comment]
I think that a different color does not help and that it makes it harder to draw mutations in other tools. Therefore, I think that a different shape is necessary.

From the shapes you suggested I like the hourglass shape most. I do not like the rhomboid since the font and the shape have a weird visual interaction and imply a different reading direction. The one sided-arrow does look more like a label to me but that may be the case because of the spacing.

**Comment by lhstrh** [comment]
It's really helpful to see these options juxtaposed. To me, it's a tie between the hourglass and the rhombus. The advantage of the latter is that it still has a forward directionality to it like the chevron does...

**Reply by cmnrd** [reply]
Yes, that is also what I like about the rhombus. But now I also see @soerendomroes's point about the interaction with the text.

**Reply by lhstrh** [reply]
I actually didn't get that point about the interaction... Help me out?

**Reply by soerendomroes** [reply]
The rhombus makes me look to the top right in the direction of the left and right border, while the font suggests a left-to-right reading direction. But maybe this is the case because the spacing between the rhombus and the text is smaller. Moreover, outgoing edges could also change this.

**Reply by cmnrd** [reply]
Ok, then I guess I missed your point. I thought you mean, that the eye kind of expects the text to be slanted similarly to the rhomboid. The spacing would need to be optimized of course. Perhaps the interaction with the text is also not too important, as we show numbers most of the time.
![Mutation](https://github.com/lf-lang/lingua-franca/assets/6460123/1f3e9957-9d44-43f8-8016-bf45f8a7eca6)

**Reply by lhstrh** [reply]
I think that what @cmnrd shows up here looks good. I don't really perceive any problem with the rhombus shape.

**Reply by axmmisaka** [reply]
Just to chime in, I like the idea of using a parallelogram for a very specific reason - the typescript runtime have direct support with Mermaid.js, which we might find handy to represent small working examples. While the reaction symbol is not representable and we need to find workarounds, future decisions could potentially take its limited shape support into account.

```mermaid
flowchart LR
    id1>This looks? like a reaction]
    id2[/this looks like a mutation/]
    id3{I shutdown}
    id4((I startup))

    id4-->id1
    id3-->id2
```
